packages/jupyterlab-lsp/src/transclusions/ipython-graph/extractors.ts

```diff
@@ -0,0 +1,36 @@
+import { IForeignCodeExtractorsRegistry } from '../../extractors/types';
+import { RegExpForeignCodeExtractor } from '../../extractors/regexp';
+
+export let foreign_code_extractors: IForeignCodeExtractorsRegistry = {
+  // general note: to match new lines use [^] instead of dot, unless the target is ES2018, then use /s
+  python: [
+    new RegExpForeignCodeExtractor({
+      language: 'sparql',
+      pattern: '^%%(sparql)( .*?)?\n([^]*)',
+      extract_to_foreign: '$3',
+      is_standalone: true,
+      file_extension: 'rq'
+    }),
+    new RegExpForeignCodeExtractor({
+      language: 'ttl',
+      pattern: '^%%(ttl)( .*?)?\n([^]*)',
+      extract_to_foreign: '$3',
+      is_standalone: false,
+      file_extension: 'ttl'
+    }),
+    new RegExpForeignCodeExtractor({
+      language: 'ttl',
+      pattern: '^%%(shacl)( .*?)?\n([^]*)',
```

Comment:
really? i might leave out the shacl one, as i don't know if their representation is really a standard...